administrator/components/com_finder/src/Indexer/Debugadapter.php

```diff
@@ -0,0 +1,979 @@
+
+		// Iterate through the items and index them.
+		for ($i = 0, $n = count($items); $i < $n; $i++)
+		{
+			// Index the item.
+			$this->index($items[$i]);
+
+			// Adjust the offsets.
+			$offset++;
+			$iState->batchOffset++;
+			$iState->totalItems--;
+		}
+
+		// Update the indexer state.
+		$aState['offset'] = $offset;
+		$iState->pluginState[$this->context] = $aState;
+		Indexer::setState($iState);
+
+		return true;
+	}
+
+	/**
+	 * Method to remove outdated index entries
+	 *
+	 * @return  integer
+	 *
+	 * @since   4.1
+	 */
+	public function onFinderGarbageCollection()
+	{
+		$db = Factory::getDbo();
+		$type_id = $this->getTypeId();
+
+		$query = $db->getQuery(true);
+		$subquery = $db->getQuery(true);
+		$subquery->select('CONCAT(' . $db->quote($this->getUrl('', $this->extension, $this->layout)) . ', id)')
+			->from($db->quoteName($this->table));
+		$query->select($db->quoteName('l.link_id'))
+			->from($db->quoteName('#__finder_links', 'l'))
+			->where($db->quoteName('l.type_id') . ' = ' . $type_id)
+			->where($db->quoteName('l.url') . ' LIKE ' . $db->quote($this->getUrl('%', $this->extension, $this->layout)))
+			->where($db->quoteName('l.url') . ' NOT IN (' . $subquery . ')');
+		$db->setQuery($query);
+		$items = $db->loadColumn();
+
+		foreach ($items as $item)
+		{
+			$this->indexer->remove($item);
+		}
+
+		return count($items);
+	}
+
+	/**
+	 * Method to change the value of a content item's property in the links
+	 * table. This is used to synchronize published and access states that
+	 * are changed when not editing an item directly.
+	 *
+	 * @param   string   $id        The ID of the item to change.
+	 * @param   string   $property  The property that is being changed.
+	 * @param   integer  $value     The new value of that property.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @throws  Exception on database error.
+	 */
+	protected function change($id, $property, $value)
+	{
+		// Check for a property we know how to handle.
+		if ($property !== 'state' && $property !== 'access')
+		{
+			return true;
+		}
+
+		// Get the URL for the content id.
+		$item = $this->db->quote($this->getUrl($id, $this->extension, $this->layout));
+
+		// Update the content items.
+		$query = $this->db->getQuery(true)
+			->update($this->db->quoteName('#__finder_links'))
+			->set($this->db->quoteName($property) . ' = ' . (int) $value)
+			->where($this->db->quoteName('url') . ' = ' . $item);
+		$this->db->setQuery($query);
+		$this->db->execute();
+
+		return true;
+	}
+
+	/**
+	 * Method to index an item.
+	 *
+	 * @param   Result  $item  The item to index as a Result object.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @throws  Exception on database error.
+	 */
+	abstract protected function index(Result $item);
+
+	/**
+	 * Method to reindex an item.
+	 *
+	 * @param   integer  $id  The ID of the item to reindex.
```

Comment:
```suggestion
	 * @param   int  $id  The ID of the item to reindex.
```

Param uses short version, return uses long version